### PR TITLE
Add exercise: acronym

### PR DESCRIPTION
**Pull request by glennj**
This implementation **does require 2201.5**. It uses RegExp, [introduced in this version](https://ballerina.io/downloads/swan-lake-release-notes/swan-lake-2201.5.0#language-support-for-regular-expressions).

### CODE REVIEW

**Review by ErikSchierboom — APPROVED** [review]

File: exercises/practice/acronym/tests/acronym_test.bal [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        "lowercase words": ["Ruby on Rails", "ROR"],
+        "punctuation": ["First In, First Out", "FIFO"],
+        "all caps word": ["GNU Image Manipulation Program", "GIMP"],
+        "punctuation without whitespace": ["Complementary metal-oxide semiconductor", "CMOS"],
+        "very long abbreviation": ["Rolling On The Floor Laughing So Hard That My Dogs Came Over And Licked Me", "ROTFLSHTMDCOALM"],
+        "consecutive delimiters": ["Something - I made up from thin air", "SIMUFTA"],
+        "apostrophes": ["Halley's Comet", "HC"],
+        "underscore emphasis": ["The Road _Not_ Taken", "TRNT"]
+    };
+}
```
Comment (ErikSchierboom) [inline exercises/practice/acronym/tests/acronym_test.bal:22]: Parameterized tests can be a bit cumbersome:

- Any error usually points to the failing assertion, which makes it slightly harder for the student to spot where the error was
- If we ever upgrade the test runner to version 2, the "test_code" field might just contain the assertion, unless there is a nice way to get the actual value
- Separate test cases are easier to solve in a TDD like way

**Review by ErikSchierboom — APPROVED** [review]

File: exercises/practice/acronym/tests/acronym_test.bal (hunk 1 above)
Comment (glennj) [reply exercises/practice/acronym/tests/acronym_test.bal:22]: Right, the test runner. I knew there was something I was forgetting.

Ballerina seems to be pretty good at saying which test case failed, even with the dataProvider pattern.

I haven't added "skips" yet, now would be a good time to start.

Comment (glennj) [reply exercises/practice/acronym/tests/acronym_test.bal]: handling skips in the exercise validation script needs to be done first.

**Comment by ErikSchierboom** [comment]
@exercism/ballerina I'll be merging some of these PRs to get some exercises merged before we're launching Mind Shifting May in a couple of days. Of course, I'll only merge them if the tests file looks good and if CI passes all the checks. If there are issues with an exercise we can do a follow-up PR to fix this.

**Pull request merged**